Request to adding UML diagram #2

Alvahaze opened the pull request:
- Re-created UML diagram  c29fe57
- Added UML diagram classes  fcad997
- Fixed some UML diagram classes  49337ec
- Fixed some UML classes  638fb80
- Fixed UML diagram  328b396
- UML architecture fixes  99f7439
- Fixed UML diagram  cf28cde
- UML diagram architecture  e6b1310
- Some fixes in UML  4edd0fb
- Small fix in UML diagram  b35cf8c

pisarevski committed:
Re-created UML diagram

- Deleted old structure
- Added package trade with containing classes
- Added class Storage
- Added interface IStorageable
- Added interface IAttackable
- Added package ship with containing classes
- Added package parts with containing classes
- Added class Universe

AlexRakitskiy committed:
Added UML diagram classes

- Added ammunition package with containing classes
- Added locations package with containing classes
- Added events package with containing classes

pisarevski committed:
Fixed some UML diagram classes

- Fixed class Universe
- Fixed class Storage
- Fixed class Hull

AlexRakitskiy committed:
Fixed some UML classes

- Fixed ILandable interface
- Fixed Gun class

AlexRakitskiy committed:
Fixed UML diagram

- Added name and size fields for Gun and Magazine classes
- Gun and Magazine now implements IStorageable interface
- Renamed field in class Gun from baseDamage to damage
- Renamed field in class Ammo from baseDamage to damageIncreasing
- Removed field AmmoType from class Ammo and AmmoType enumerator
- Added new classes extends from Location
- Added new class Battle extends from Event

pisarevski committed:
UML architecture fixes

- Fixes in methods (class Universe)
- Changes in Storage class
- Changes in class Starship (added hitPoints)
- Changes in interface IStorageable
- Changes in class Trader

AlexRakitskiy committed:
Fixed UML diagram

- Fixed Battle class

pisarevski committed:
UML diagram architecture

- Added constructor for class Event
- Added constructor for class Starship
- Added storage in class OpenSpace
- Added package stuff
- Added class MedicineChest
- Added fields and methods for class MedicineChest
- Little fix in class storage
- Little fix in class class Mechanic
- Little fix in class AverageTrader
- Deleted class Battle

Alvahaze commented:
This is a new pull request. Request #9 is no longer relevant.

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,187 @@
+    class Universe {
+        -int kills;
+        -int moneySpends;
+        +void startUniverse();
```
Maybe `journeyState()`?

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,187 @@
 
-        class PirateStarship {
-            -Turret rightTurret;
+            +void Event(Universe universe, Location location, GitStarship gitStarship);
```
Event may not contain a 'location'.

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,187 @@
-        class PirateStarship {
-            -Turret rightTurret;
+            +void Event(Universe universe, Location location, GitStarship gitStarship);
+            +void start(GitStarship gitStarship);
```
If `gitStarship` assigned via the constructor, it is a unnesessary to pass it through `start()` parameter.

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,187 @@
+        Starship <|-- PirateStarship
 
-        class Planet {
+        abstract class Starship implements IAttackable {
```
Your ships can't land to 'ILandable'.

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,187 @@
-            -int value;
+        +class OpenSpace extends Location {
+            -ArrayList<Starship> ships;
+            -Storage storage;
```
Now, if all subclasses of `Location` has `storage`, why `storage` is not presented in base class?

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,187 @@
-        -HashMap<String, IStoreable> iStoreableHashMap;
+    package stuff {
+
+        +class MedicineChest implements IStorageable {
```
How this class works?

Alvahaze replied:
Fixed! Delete this package.

pisarevski committed:
Some fixes in UML

- Renamed method startUniverse in jorneyState in the class Universe
- Delete argument location in constructor in class Event
- Method start in class Event no more contain arguments
- Field storage now in abstract class Location
- Starship now implements ILandable interface
- Delete package stuff and included classes

Alvahaze commented:
Ok, I add some fixes in UML? please review again!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,174 @@
-
-        ValuableItem <|-- TurretMagazineCase
-        ValuableItem <|-- MetalShell
+    interface ILandable {
```
Unclear interface using.

pisarevski committed:
Small fix in UML diagram

- Class Starship is no more implements ILandable

Alvahaze commented:
I has made some fixes in UML! please review again!

ViRGiL175 commented:
Approved.
